cli/src/cli_util.rs

```diff
@@ -1281,6 +1294,24 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
             writeln!(ui.status(), "Rebased {num_rebased} descendant commits")?;
         }
 
+        for (workspace_id, wc_commit_id) in tx.mut_repo().view().wc_commit_ids().clone() {
+            if self
+                .check_repo_rewritable(tx.mut_repo(), [&wc_commit_id])
+                .is_err()
+            {
+                let wc_commit = tx.repo().store().get_commit(&wc_commit_id)?;
+                tx.mut_repo()
+                    .check_out(workspace_id.clone(), &self.settings, &wc_commit)?;
+                if workspace_id == self.workspace_id().clone() {
+                    writeln!(
+                        ui.warning_default(),
+                        "The working-copy commit became immutable, so a new commit has been \
+                         created on top of it."
+                    )?;
+                }
```

Comment:
I think it makes sense to print a message for other workspaces too. Maybe "The working-copy commit in workspace {workspace_id} became immutable, ..."?